metagraph/src/annotation/taxonomy/taxo_classifier.cpp

```diff
@@ -65,7 +65,7 @@ void TaxoClassifier::update_scores_and_lca(const TaxId start_node,
                                            tsl::hopscotch_map<TaxId, uint64_t> &node_scores,
                                            tsl::hopscotch_set<TaxId> &nodes_already_propagated,
                                            TaxId &best_lca,
-                                           uint64_t &best_lca_dist_to_root) {
+                                           uint64_t &best_lca_dist_to_root) const {
```

Comment:
If you modify a parameter, pass it by pointer
```suggestion
                                           uint64_t *best_lca_dist_to_root) const {
```

Reply:
will do the changes, thanks! I remember that you mention exactly this in an another comment last week, but at that time I turned to pointer only the parameters which were created inside the method. It is clear now and will address all the comments after fixing the code

Reply:
I modified eveywhere to pass pointers when modifying parameters, thanks! The only thing that is not working is in `tax_classify` CLI where it looks that if I send the results (`vector<pair<string, string>>`) as a pointer, then the multithreading crashes. I tried to debug it, but i couldn't figure it out the reason. Here is a small PR with the changes https://github.com/ratschlab/metagraph/pull/311/files    Do you have any idea why is this happening?